subprojects/docs/src/snippets/java/customDirs/kotlin/build.gradle.kts

```diff
@@ -55,9 +55,17 @@ reporting.baseDir = file("my-reports")
 project.setProperty("testResultsDirName", "$buildDir/my-test-results")
 
 tasks.register("showDirs") {
+    val rootDirPath = project.rootDir.toPath()
+
+    val reportsDir: Directory = project.the<ReportingExtension>().getBaseDirectory().get()
+    val reportsDirPath = reportsDir.getAsFile().toPath()
+
+    val testResultsDir: Directory = project.the<JavaPluginExtension>().getTestResultsDir().get()
```

Comment:
🔴 It's better to delay the call to `get()` since, in theory, the value of the property could still change after this task has been configured.